From 0dcf6081140b7fa20e9946b33b957fc6ced88629 Mon Sep 17 00:00:00 2001
From: "dk.han" <dk.han@lge.com>
Date: Tue, 20 Jan 2015 11:00:42 +0900
Subject: [PATCH 62/95] Fix bugs don't remove track of DirectOutputThread

Don't remove track of DirectOutputThread by a wrong state control for track stop processor.
Change a state to STOPPING_1 in Track::stop() function, but check STOPPING_2 state in DirectOutput::prepareTracks_l
function.
So the Track is doesn't add tracksToRemove vector.

I thing, change check condition from isStopping_2() to isStopping() function in DirectOutput::prepareTracks_l() function.

Change-Id: I68c4346f02ef1b8635a2180c2a0b3089ce2af955
Signed-off-by: DongKyoon Han <dk.han@lge.com>
---
 services/audioflinger/Threads.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/services/audioflinger/Threads.cpp b/services/audioflinger/Threads.cpp
index 6b23f0f..670d20d 100644
--- a/services/audioflinger/Threads.cpp
+++ b/services/audioflinger/Threads.cpp
@@ -4069,7 +4069,7 @@ AudioFlinger::PlaybackThread::mixer_state AudioFlinger::DirectOutputThread::prep
         }
 
         if ((track->framesReady() >= minFrames) && track->isReady() && !track->isPaused() &&
-                !track->isStopping_2() && !track->isStopped())
+                !track->isStopping() && !track->isStopped())
         {
             ALOGVV("track %d s=%08x [OK]", track->name(), cblk->mServer);
 
-- 
2.5.0

